core/jobs/batch_jobs/mailchimp_population_jobs.py

```diff
@@ -0,0 +1,188 @@
+                before this function in the beam job filtered for the same.
+            batch_index_dict: Dict. Single element dictionary of current batch
+                index.
+
+        Raises:
+            Exception. Exception thrown by the api is raised.
+
+        Yields:
+            JobRunResult. Job run result which is either 'Ok' or an error with
+            corresponding error message.
+        """
+        client = mailchimp3.MailChimp(    # pragma: no cover
+            mc_api=feconf.MAILCHIMP_API_KEY, mc_user=feconf.MAILCHIMP_USERNAME)
+
+        email_tuples = email_tuples[0][
+            batch_index_dict['batch_index_for_mailchimp'] * 500:
+            (batch_index_dict['batch_index_for_mailchimp'] + 1) * 500]
+        mailchimp_data = []
+
+        for email, _ in email_tuples:
+            mailchimp_data.append({
+                'email_address': email[0],
+                'status': 'subscribed'
+            })
+
+        try:
+            response = client.lists.update_members(
+                feconf.MAILCHIMP_AUDIENCE_ID,
+                {'members': mailchimp_data, 'update_existing': True})
+        except mailchimpclient.MailChimpError as error:
+            error_message = ast.literal_eval(str(error))
+            raise Exception(error_message['detail']) from error
+
+        if (
+                len(response['new_members']) + len(response['updated_members'])
+                == len(email_tuples)):
+            yield result.Ok()
+        else:
+            failed_emails = []
+            for user in response['errors']:
+                failed_emails.append(user['email_address'])
+            yield result.Err('User update failed for: %s' % failed_emails)
+
+
+class CombineTuples(beam.CombineFn):  # type: ignore[misc]
+    """CombineFn for combining all user email, status tuples."""
+
+    def create_accumulator(self) -> List:
+        """Base accumulator where the tuples are added."""
+        return []
+
+    def add_input(self, accumulator: List, model: Tuple) -> List:
+        """Append each tuple to the accumulator list."""
+        accumulator.append(model)
+        return accumulator
+
+    def merge_accumulators(self, accumulators: Iterable[List]) -> List:
+        """Merging accumulators is just combining both of them into a single
+        list.
+        """
+        return list(accumulators)
+
+    def extract_output(
+        self, accumulator: List
+    ) -> List:
+        """Output is the accumulator itself."""
+        return accumulator
+
+
+class MailchimpPopulateJob(base_jobs.JobBase):
+    """One-off job for populating the mailchimp db."""
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        # Pcollection that returns the relevant config property with batch
+        # index.
+        config_property = (
+            self.pipeline
+            | 'Get all config properties' >> ndb_io.GetModels(
+                config_models.ConfigPropertyModel.get_all())
```

Comment:
I think you might be able to modify this query so that it returns just the model with one ID.

Reply:
I was able to modify it below for `site_updates == True`, but here whenever I try `.query(config_models.ConfigPropertyModel.id == 'batch_index_for_mailchimp')` , I get the error `cannot filter a non-Node argument, received False`. Does it not work for IDs?

Reply:
IDs work differently, I think, since they aren't defined on the model itself (see BaseModel). Any reason not to use get_by_id() or similar rather than a query?

(For keys I think it's something like what's described here: https://stackoverflow.com/a/10179686.)